File: 538. Convert BST to Greater Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function(root) {
    let sum = 0;
    loop(root);
    return root;
    
        function loop(root) {
        if (root) {    
            loop(root.right);
            let node = root.val;
            root.val = node + sum;
            sum = root.val;
            loop(root.left);
        }
    };
}

File: 538. Convert BST to Greater Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        def convert(root):
            if not root:
                return None
            convert(root.right)
            self.sm += root.val
            root.val = self.sm
            convert(root.left)
            return root
        
        self.sm = 0
        return convert(root)
